src/main/java/io/github/darealturtywurty/ancientology/core/util/Color.java

```diff
@@ -0,0 +1,955 @@
+            f = new float[3];
+        } else {
+            f = compArray;
+        }
+        if (frgbvalue == null) {
+            f[0] = (getRed()) / 255f;
+            f[1] = (getGreen()) / 255f;
+            f[2] = (getBlue()) / 255f;
+        } else {
+            f[0] = frgbvalue[0];
+            f[1] = frgbvalue[1];
+            f[2] = frgbvalue[2];
+        }
+        return f;
+    }
+
+    /**
+     * Returns a {@code float} array containing the color and alpha components of
+     * the {@code Color}, in the {@code ColorSpace} of the {@code Color}. If
+     * {@code compArray} is {@code null}, an array with length equal to the number
+     * of components in the associated {@code ColorSpace} plus one is created for
+     * the return value. Otherwise, {@code compArray} must have at least this length
+     * and it is filled in with the components and returned.
+     *
+     * @param  compArray an array that this method fills with the color and alpha
+     *                   components of this {@code Color} in its {@code ColorSpace}
+     *                   and returns
+     * @return           the color and alpha components in a {@code float} array.
+     */
+    public float[] getComponents(float[] compArray) {
+        if (fvalue == null) { return getRGBComponents(compArray); }
+        float[] f;
+        int n = fvalue.length;
+        if (compArray == null) {
+            f = new float[n + 1];
+        } else {
+            f = compArray;
+        }
+        for (int i = 0; i < n; i++) {
+            f[i] = fvalue[i];
+        }
+        f[n] = falpha;
+        return f;
+    }
+
+    /**
+     * Returns a {@code float} array containing only the color components of the
+     * {@code Color}, in the {@code ColorSpace} of the {@code Color}. If
+     * {@code compArray} is {@code null}, an array with length equal to the number
+     * of components in the associated {@code ColorSpace} is created for the return
+     * value. Otherwise, {@code compArray} must have at least this length and it is
+     * filled in with the components and returned.
+     *
+     * @param  compArray an array that this method fills with the color components
+     *                   of this {@code Color} in its {@code ColorSpace} and returns
+     * @return           the color components in a {@code float} array.
+     */
+    public float[] getColorComponents(float[] compArray) {
+        if (fvalue == null) { return getRGBColorComponents(compArray); }
+        float[] f;
+        int n = fvalue.length;
+        if (compArray == null) {
+            f = new float[n];
+        } else {
+            f = compArray;
+        }
+        for (int i = 0; i < n; i++) {
+            f[i] = fvalue[i];
+        }
+        return f;
+    }
+
+    public static Color random(Random rand) {
+        float r = rand.nextFloat() / 2f;
+        float g = rand.nextFloat() / 2f;
+        float b = rand.nextFloat() / 2f;
+        return new Color(r, g, b);
+    }
+
+}
```

Comment:
I don't really see the point of this. Just a waste of space. It's much easier and more readable to just use hexadecimal colour codes where colour is used.